src/Form/FormControlDecoratorGroup.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useFormGroupContext } from './FormGroupContext';
+import { FORM_CONTROL_SIZES } from './constants';
+import FormControlFloatingLabel from './FormControlFloatingLabel';
+import FormControlDecorator from './FormControlDecorator';
+
+/**
+  * Decorates a textual input.
+  */
+const FormControlDecoratorGroup = ({
+  children,
+  leadingElement,
+  trailingElement,
+  floatingLabel,
+  className,
+  ...props
+}) => {
+  const { size } = useFormGroupContext(props);
+  return (
+    <div
+      className={classNames(
+        'pgn__form-control-decorator-group',
+        {
+          'has-prepended-node': !!leadingElement,
+          'has-appended-node': !!trailingElement,
+          'has-leading-element': !!leadingElement,
+          'has-trailing-element': !!trailingElement,
+          'has-floating-label': !!floatingLabel,
+          'pgn__form-control-decorator-group-lg': size === FORM_CONTROL_SIZES.LARGE,
+          'pgn__form-control-decorator-group-sm': size === FORM_CONTROL_SIZES.SMALL,
+        },
+        className,
+      )}
+      {...props}
+    >
+      {children}
+      {leadingElement && <FormControlDecorator location="leading">{leadingElement}</FormControlDecorator>}
+      {trailingElement && <FormControlDecorator location="trailing">{trailingElement}</FormControlDecorator>}
+      {floatingLabel && <FormControlFloatingLabel>{floatingLabel}</FormControlFloatingLabel>}
+    </div>
+  );
+};
+
+FormControlDecoratorGroup.propTypes = {
+  children: PropTypes.node.isRequired,
+  leadingElement: PropTypes.node,
```

Comment:
I think I asked this before but I don't remember the answer, what happens if several of these (`leadingElement`, `trailingElement`) are true at the same time?

Reply:
These are elements are slotted in containers so if they are multiple nodes they'll stack horizontally. It works, but should be avoided until the form control decorator is measuring their widths. Right now we assume with the width of these elements roughly matches the width of an icon button. If we want to support anything in here, we should measure the size of the rendered objects and update the horizontal padding of the underlying form control to ensure its content doesn't run underneath.